docs/topics/kotlin-language-features-and-proposals.md

```diff
@@ -0,0 +1,384 @@
+
+* **KEEP discussion**. We are fairly certain that the feature should be added to the language.
+  We aim to provide a motivation, use-cases, design, and other important details in a document called a _KEEP_. 
+  We expect feedback from users to focus on discussing all the information provided in the KEEP.
+
+* **In preview**. A feature prototype is ready, and you can enable it using a feature-specific compiler option.
+  We seek feedback on your experience with the feature, including how easily it integrates into your codebase,
+  how it interacts with existing code, and any IDE support issues or suggestions.
+  The feature's design may change significantly, or it could be completely revoked based on feedback. When a feature is 
+  _in preview_, it has a [stability level](components-stability.md#stability-levels-explained).
+
+* **Stable**. The language feature is now a first-class citizen in the Kotlin language.
+  We guarantee its backward compatibility and that we'll provide tooling support.
+
+* **Revoked**. We have revoked the proposal and will not implement the feature in the Kotlin language.
+  We may revoke a feature that is _in preview_ if it is not a good fit for Kotlin.
+
+## List of Kotlin language features and design proposals
+
+> This table contains Kotlin language features and proposals starting from Kotlin 1.7.0.
+>
+{type="note"}
+
+<tabs>
+<tab id="all-proposals" title="All">
+
+<!-- <include include-id="all-proposals" src="all-proposals.xml"/> -->
+
+<chunk id="source">
+<table header-style="left">
+
+<!-- EXPLORATION AND DESIGN BLOCK -->
+
+<tr filter="exploration-and-design">
+<td>
+
+**Exploration and design**
+
+</td>
+<td>
+
+**Kotlin statics and static extensions**
+
+* KEEP proposal: [statics.md](https://github.com/Kotlin/KEEP/blob/statics/proposals/statics.md)
+* YouTrack issue: [KT-11968](https://youtrack.jetbrains.com/issue/KT-11968)
+
+</td>
+</tr>
+
+<tr filter="exploration-and-design">
+<td>
+
+**Exploration and design**
+
+</td>
+<td>
+
+**Collection literals**
+
+* KEEP proposal: [collection-literals.md](https://github.com/KyLeggiero/KEEP/blob/collection-literals/proposals/collection-literals.md)
+* YouTrack issue: [KT-43871](https://youtrack.jetbrains.com/issue/KT-43871)
+
+</td>
+</tr>
+
+<tr filter="exploration-and-design">
+<td>
+
+**Exploration and design**
+
+</td>
+<td>
+
+**Union types for errors and exceptions**
+
+* KEEP proposal: Not defined
+* YouTrack issue: [KT-7128](https://youtrack.jetbrains.com/issue/KT-7128)
+
+</td>
+</tr>
+
+<tr filter="exploration-and-design">
+<td>
+
+**Exploration and design**
+
+</td>
+<td>
+
+**Name-based destructuring**
+
+* KEEP proposal: Not defined
+* YouTrack issue: [KT-19627](https://youtrack.jetbrains.com/issue/KT-19627)
+
+</td>
+</tr>
+
+<tr filter="exploration-and-design">
+<td>
+
+**Exploration and design**
+
+</td>
+<td>
+
+**Support immutability (multifield value classes)**
+
+* KEEP proposal: Not defined
```

Comment:
Let's add this: KEEP notes: [immutability](https://github.com/Kotlin/KEEP/blob/master/notes/value-classes.md#immutability-and-value-classes)